### PR TITLE
Added support for kraken2

### PR DESCRIPTION
**Pull request by charles-cowart**
Added support for kraken2. Multiple libraries are supported for minimap2 as well.

### CODE REVIEW

File: sequence_processing_pipeline/QCJob.py [hunk 1]
```diff
@@ -411,18 +405,41 @@ def _gen_chained_cmd(self, current_dir, filename1, filename2,
...
             result += (f' --adapter_sequence {adapter_a}'
                        f' --adapter_sequence_r2 {adapter_A}')
 
-        result += (f' -l 100 -i {read1_input_path} -I {read2_input_path} '
-                   f'-w {self.nprocs} -j {json_output_path} -h '
-                   f'{html_output_path} --stdout | {self.minimap2_path} -ax'
-                   f' sr -t {self.nprocs} {self.mmi_db_path} - -a | '
-                   f'{self.samtools_path} fastq -@ {self.nprocs} -f 12 -F '
-                   f'256 -1 {path1} -2 {path2}')
+        result += (f' -l 100 -i {read1_input_path} -I {read2_input_path} -w '
```
Comment (antgonza) [inline sequence_processing_pipeline/QCJob.py:425]: @charles-cowar, could you open an issue to define the min length of the reads via the config file or somewhere else that is not hardcoded? Thank you.

File: sequence_processing_pipeline/tests/test_QCJob.py [hunk 2]
```diff
@@ -685,27 +690,17 @@ def test_command_chaining(self):
...
-                        print("FIRST LINE EXPECTED ##################")
-                        print(config['first_line'])
-                        print("##################")
-                    if lines_obs[-1] != config['last_line']:
-                        print("LAST LINE OBSERVED ##################")
-                        print(lines_obs[-1])
-                        print("LAST LINE EXPECTED ##################")
-                        print(config['last_line'])
-                        print("##################")
+                print("LAST LINE: %s" % lines_obs[-1])
```
Comment (antgonza) [inline sequence_processing_pipeline/tests/test_QCJob.py:693]: I think is fine to also remove the last print, no?

**Review by antgonza — CHANGES_REQUESTED** [review]
Thank you @charles-cowart, looks good, a few question and I _think_ we are missing a test for when the minimap2 command has more than 2 references, no? ... The test can be a direct call to _gen_chained_cmd, just to make sure that they are generated fine and we don't mess something up with future changes.

File: sequence_processing_pipeline/QCJob.py (hunk 1 above)
Comment (charles-cowart) [reply sequence_processing_pipeline/QCJob.py:425]: Done! https://github.com/biocore/mg-scripts/issues/76

File: sequence_processing_pipeline/tests/test_QCJob.py (hunk 2 above)
Comment (charles-cowart) [reply sequence_processing_pipeline/tests/test_QCJob.py]: definitely. done!

**Comment by charles-cowart** [comment]
@antgonza unitest created for supplying multiple databases to minimap2 and kraken2

**Review by antgonza — APPROVED** [review]
Looks good, thank you.

**Pull request merged**